patches/tModLoader/Terraria/Item.cs.patch

```diff
@@ -1813,7 +1814,7 @@
  		if (Main.rand == null)
  			Main.rand = new UnifiedRandom();
  
-+		if (NPCLoader.blockLoot.Contains(Type))
++		if (NPCLoader.blockLoot.Contains(Type) || newItemDisabled)
```

Comment:
Should we consider generalizing the `NPCLoader.blockLoot` hack in a followup PR? It could probably also benefit from a `using var` thing at least. We might be able to combine the two features, so that modders can also block vanilla tile drops individually in `GlobalItem.Drop` with something like `Item.BlockVanillaDropsOf(type)` or something